### PR TITLE
Develop jQuery

### PR DESCRIPTION
**Pull request by AnastasiyaKislach**

### CODE REVIEW

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+﻿(function ($) {
+
+    "use strict";
+
+    var data = [],
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:5]: You can't apply this plugin to multiple elements on the same page with variables in plugin's global scope.

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            self = this;
+            self
+                .on("dragenter",
+                    function (e) {
+                        e.currentTarget.style.borderColor = "#61696C";
+                    })
+                .on("drop",
+                    function(e) {
+                        e.preventDefault();
+                        e.currentTarget.style.border = "";
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:19]: You should use
```
$.css({
    borderColor: "#61696C
});
```

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                continue;
+            }
+
+            var reader = new FileReader();
+
+            reader.onload = (
+                function (e) {
+                    var selector = $(".photos");
+                    if (!selector.length) {
+                        $('#' + targetSelector.id).after(function () {
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:56]: `$(targetSelector)` should work. What if _targetSelector_ el doesn't have id?

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                            return '<div class="photos"></div>';
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:57]: You should create new elements using jQuery:
```
$(targetSelector).after($('<div>',{
   class: 'photos'
}))
```

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                        });
+                        selector = $(".photos");
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:59]: It's good practice to prepend jQuery variables with $ i.e. $selector

Comment (DenomikoN) [reply 5/DragAndDropDemo/App/drag-and-drop.js:59]: What if there any other elements with .photos class? You can store the one you created in a variable and get rid of this selector.

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+                    }
+                    selector.append("<img src='" + e.target.result + "' class='img-responsive image' alt='" + filename + "'/>");
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:61]: Create elements using jQuery:
```
$selector.append($('<img>',{
   src: e.target.result,
   class: 'img-responsive image',
   alt: filename
}));
```

File: 5/DragAndDropDemo/App/drag-and-drop.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        getData: function() {
+            return data;
+        }
+    };
+
+    function handlerUploads(files, targetSelector) {
+        var length = files.length;
+
+        for (var i = 0; i < length; i++) {
+            let f = files[i],
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/drag-and-drop.js:43]: f is not a meaningful variable name

File: 5/DragAndDropDemo/App/script.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            } else {
+                $.ajax('http://jsonplaceholder.typicode.com/posts', {
+                    method: 'POST',
+                    data: data,
+                    success: function (response) {
+                        console.log(response);
+                        loadFiles(global.draganddrop);
+                    }
+                });
+                resolve("Worked!");
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/script.js:60]: It's probably should be:
```
$.ajax(...)
.done(function() {
    resolve('Worked!');
  })
.fail(function() {
     reject("Error!");
})
```

File: 5/DragAndDropDemo/App/script.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            }
+        });
+    }
+
+    $(document).ready(function () {
+        global.draganddrop = $("div.draganddrop");
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App/script.js:66]: Why do you need to keep it in a global variable?

File: 5/DragAndDropDemo/App_Start/WebApiConfig.cs [hunk 10]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Web.Http;
+
+namespace DragAndDropDemo
```
Comment (DenomikoN) [inline 5/DragAndDropDemo/App_Start/WebApiConfig.cs:3]: :+1: for using WebAPI.

File: 5/LazyLoadingDemo/App/lazy-load.js [hunk 11]
```diff
@@ -0,0 +1,104 @@
+﻿(function ($) {
+
+    "use strict";
+
+    var self;
```
Comment (DenomikoN) [inline 5/LazyLoadingDemo/App/lazy-load.js:5]: Again variables in plugin's global scope.

File: 5/LazyLoadingDemo/App/lazy-load.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+            albumId: image.albumId,
+            id: 'imgLazy_' + image.id,
+            url: image.url,
+            alt: image.title,
+            'class': 'trambnail'
+
+        });
+    }
+
+    function proportion() {
```
Comment (DenomikoN) [inline 5/LazyLoadingDemo/App/lazy-load.js:82]: Is it used anywhere?

File: 5/LazyLoadingDemo/App/lazy-load.js [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+        width: "590px",
+        height: "500px",
+        min_height: 150
+    };
+
+
+    var methods = {
+        init: function (options) {
+            if (options) {
+                $.extend(settings, options);
```
Comment (DenomikoN) [inline 5/LazyLoadingDemo/App/lazy-load.js:21]: it should look like:
var settings = $.extend({}, defaultSettings, options);

File: 5/LazyLoadingDemo/App/lazy-load.js [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+                $.ajax({
+                    method: "GET",
+                    url: settings.resource,
+                    dataType: "json",
+                    success: function (response) {
+                        settings.images = response;
+                        proportionVal = Math.floor(proportion());
+                        hiddenImgs = settings.images;
+                        showVisibleImages();
+                        $(self)
```
Comment (DenomikoN) [inline 5/LazyLoadingDemo/App/lazy-load.js:37]: you can store it in a variable once initializing plugin

File: 5/LazyLoadingDemo/App/lazy-load.js [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+        var lenght = hiddenImgs.length;
+        if (proportionVal > lenght) {
+            proportionVal = lenght;
+        }
+        var i = 0;
+        for (i = 0; i < proportionVal; i++) {
+            let template = createTemplate(hiddenImgs[i]);
+            self.append(template);
+
+            $('#imgLazy_' + hiddenImgs[i].id).on('click', null, { src: hiddenImgs[i].url }, showFullImage);
```
Comment (DenomikoN) [inline 5/LazyLoadingDemo/App/lazy-load.js:98]: Why not simply `$(hiddenImgs[i])`

**Review by DenomikoN — APPROVED** [review]
Задание засчитываю. Не забывай, когда работаешь с jQuery использовать его на всю. + подумай как объявлять плагин чтобы его можно было использовать более чем один раз на странице.

File: 5/LazyLoadingDemo/App/lazy-load.js (hunk 12 above)
Comment (AnastasiyaKislach) [inline 5/LazyLoadingDemo/App/lazy-load.js]: Yes. It's used within ajax success function.

**Pull request merged**